packages/ag-charts-community/src/chart/dom/domManager.ts

```diff
@@ -245,6 +247,24 @@ export class DOMManager extends BaseManager<Events['type'], Events> {
         return this.rootElements['canvas'].element.getBoundingClientRect();
     }
 
+    getDocumentRoot() {
+        const docRoot = getDocument('body');
+        let parent = this.container;
+
+        while (parent != null) {
+            if (parent === docRoot) {
+                return docRoot;
+            }
+            if (parent.parentElement == null) {
+                return parent;
+            }
+
+            parent = parent.parentElement as HTMLElement;
+        }
+
+        return docRoot;
+    }
```

Comment:
nit: `this.container.ownerDocument.body`

Reply:
Good spot

Reply:
Unfortunately `ownerDocument` is not set to the right element in the shadow-DOM case, so we still need to traverse the parents to find the closest enclosing element that is part of the shadow-DOM root.